Sources/ArcGISToolkit/Components/Offline/OnDemand/OnDemandMapAreaSelectorView.swift

```diff
@@ -0,0 +1,310 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct OnDemandMapAreaSelectorView: View {
+    /// The maximum size of the area selector view.
+    @State private var maxRect: CGRect = .zero
+    
+    /// A Binding to the CGRect of the selected map area.
+    @Binding var selectedMapRect: CGRect
+    
+    /// The bounding rectangle for the area selector view.
+    @State private var boundingRect: CGRect = .zero
+    
+    /// The inset rectangle for the area selector view.
+    @State private var insetRect: CGRect = .zero
```

Comment:
The name of this and the doc are not descriptive of what this rect is actually for. It looks like it's not an "inset" rect but actually an "outset" rect that is for positioning the rounded corners. If that's the case, it probably doesn't need to exist as a state variable at all. And if it does need to exist as a state variable, then I think it should be called something that describes what it is for. Like handlesRect or something.